cypress-io/cypress
# fix: duplicate and expired cookies

- Closes #25174
- Closes #25205
- Closes #25495

### Additional details
After introducing Cypress v12 with session/origin GA, we introduced a few cookie issues that become more broadly discoverable. These issues can be boiled down to two things
* Duplicating cookies stored in the server side cookie jar with a prepended dot on the domain. This was being caused when the server jar cookies, which are sometimes duplicated references, was being resynced through CDP with `hostOnly` being falsy, which would create the cookie under a different domain with the prepended dot. The proposed fixed for this is that cookies that are stored within the server side cookie jarm when synced back through CDP, keep their same domain. This should keep overwriting behavior consistent with the specific cookie in question, as well as pervent duplication. 
* Cookies not expiring properly. This was fairly noticeable when users would try to remove a cookie by setting a cookie with a date in the past. Instead, the cookie would just get re set to the new value, which was often `deleted`. Our cookie patch was not handling this case, nor was our propagation for expiry times from `0` epoch, which register in tough cookie as `-Infinity`. This wasn't serialized to the automated backend, causing the expiry time to be nonexistant. 

### Steps to test
I have added a few tests to a new driver test, called `cookie_misc`, that tests the duplicate cookie behavior and expiry times. If you check out the first commit of this PR, you will see these tests fail. With the branch fully checked out, tests should now pass with the expected behavior. 

### How has the user experience changed?


### PR Tasks


- [ ] ~Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)~ (No longer applicable with https://github.com/cypress-io/cypress/commit/0acdd2c7904da7f4938627d808604d5bf2f97714)
- [ ] (N/A) Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] (N/A) Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
